portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,47 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// the indexes recording the picture 
+// the slides in gallery.html is showing
+var slide1 = 0;
+var slide2 = 0;
+
 /**
- * Adds a random greeting to the page.
+ * Controls(flips) both slideshow in gallery.html
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function flip(/* num of pages to flip= */ num, 
+  /* index of the slide= */ sIndex) {
+  var slide;
+  if (sIndex == 1) {
+    slide1 += num;
+    slide = document.getElementById("slide1").getElementsByTagName("img");
+    slide1 = updateFlip(slide1, slide);
+  } else {
+    slide2 += num;
+    slide = document.getElementById("slide2").getElementsByTagName("img");
+    slide2 = updateFlip(slide2, slide);
+  }
+}
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+/**
+ * The helper function for flip(n,s); 
+ * sets the input slide to show 
+ * picture corresponding to the 'show' index
+ */
+function updateFlip(/* page index= */ show, slide) {
+  if (show > 4) {
+      show = 0;
+  } else if (show < 0) {
+      show = 4;
+  }
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  for (i = 0; i < 5; i++) {
```

Comment:
This structure is a little confusing. If you only show one (or maybe looks like two) slide at a time, could you just remember which one is displayed to hide rather than doing all these potentially redundant setting of slides to be hidden?